### PR TITLE
[awscontainerinsightreceiver] Add ECS info to collect ECS EC2 instance metrics

### PR DESCRIPTION
**Pull request by JohnWu20**
**Description:** <Describe what has changed.>

The last pr for ecs ec2 instance level metrics:
1, Add  CPU mem reserved metrics which get from cgroup.
2, Decorate the cadvisor metrics which ECS info which get from ecs task info and container info.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>
Unit test and integration test

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go [hunk 1]
```diff
@@ -152,6 +162,30 @@ func New(containerOrchestrator string, hostInfo hostInfo, logger *zap.Logger, op
 	return c, nil
 }
 
+// NewWithECSInfo creates a Cadvisor struct which can generate metrics from embedded cadvisor lib and generate metrics about ECS info
+func NewWithECSInfo(containerOrchestrator string, hostInfo hostInfo, logger *zap.Logger, ecs ecsInfo, options ...Option) (*Cadvisor, error) {
```
[the remark this answers is not in the document]
Comment (JohnWu20) [reply receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go:166]: Sure, I will use Option function.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by JohnWu20** [comment]
If there is no further review or any questions, can we get this PR merged?

**Comment by hossain-rayhan** [comment]
Hi @bogdandrutu @tigrannajaryan can we get this merged?

**Comment by alolita** [comment]
@mxiamxia can you please also review so that we have 2 reviews before I mark this ready-to-merge. Thx.

@open-telemetry/collector-approvers can one of you please review too.

**Review by mxiamxia — APPROVED** [review]

**Pull request merged**